src/modules/IscsiClientLib.rb

```diff
@@ -418,17 +418,28 @@ def saveConfig(user_in, pass_in, user_out, pass_out)
       tmp_conf = deep_copy(@config)
       tmp_val = Ops.get_list(tmp_conf, "value", [])
 
-      if Ops.greater_than(Builtins.size(user_in), 0) &&
-          Ops.greater_than(Builtins.size(pass_in), 0)
-        tmp_val = setOrAdd(tmp_val, "node.session.auth.username", user_in)
-        tmp_val = setOrAdd(tmp_val, "node.session.auth.password", pass_in)
+      if !user_in.empty? && !pass_in.empty?
+        tmp_val = setOrAdd(
+          tmp_val,
+          "discovery.sendtargets.auth.authmethod",
+          "CHAP"
+        )
+        tmp_val = setOrAdd(
+          tmp_val,
+          "discovery.sendtargets.auth.username_in",
+          user_in
+        )
+        tmp_val = setOrAdd(
+          tmp_val,
+          "discovery.sendtargets.auth.password_in",
+          pass_in
+        )
       else
-        tmp_val = delete(tmp_val, "node.session.auth.username")
-        tmp_val = delete(tmp_val, "node.session.auth.password")
+        tmp_val = delete(tmp_val, "discovery.sendtargets.auth.username_in")
```

Comment:
From the bug, https://bugzilla.suse.com/show_bug.cgi?id=978183 , I don't understand whether it is possible to have only one of user_in, pass_in empty.
If one can be empty, then this else block is wrong. If both need to be (un)specified, then the conditions can be simplified for readability (`specified = !user_out.empty? && !pass_out.empty?`)